### PR TITLE
register iterator

### PR DESCRIPTION
**Pull request by burrbull**

### CODE REVIEW

File: src/generate/register.rs [hunk 1]
```diff
@@ -1101,10 +1101,10 @@ fn lookup_in_peripheral<'p>(
     base_register: &'p str,
     base_field: &str,
     base_evs: &str,
-    all_registers: &[&'p Register],
+    mut all_registers: impl Iterator<Item = &'p Register>,
```
Comment (therealprof) [inline src/generate/register.rs:1104]: Hm, why is this `mut` now?

Comment (burrbull) [reply src/generate/register.rs:1104]: Because iteration is mutable one-off operation.
If it is harder to read, I can bring vector and slice back.

Comment (therealprof) [reply src/generate/register.rs:1104]: Hm, don't we have a concrete borrowable type for all Registers to pass in? I would expect that can build something like a `RegisterMap` and pass a reference to it around allowing called functions to obtain their own `iter` to it.

Comment (burrbull) [reply src/generate/register.rs]: I've restored previous behavior.

**Review by therealprof — APPROVED** [review]
LGTM

bors r+

**Pull request merged**